packages/docusaurus-plugin-content-blog/src/blogUtils.ts

```diff
@@ -7,11 +7,7 @@ import {parse, normalizeUrl} from '@docusaurus/utils';
 import {LoadContext} from '@docusaurus/types';
 
 export function truncate(fileString: string, truncateMarker: RegExp | string) {
-  const truncated =
-    typeof truncateMarker === 'string'
-      ? fileString.includes(truncateMarker)
-      : truncateMarker.test(fileString);
-  return truncated ? fileString.split(truncateMarker)[0] : fileString;
+  return fileString.split(truncateMarker, 1).shift() || fileString;
```

Comment:
Would it not be the same result if write just like that?

```suggestion
  return fileString.split(truncateMarker, 1).shift();
```

Reply:
Actually this is true but typescript complains shift() can return undefined. >.<

Reply:
fileString.split(truncateMarker, 1). is interpreted as `string[]`

and shift() return `string | undefined`.

Reply:
another solution is to use
```ts
fileString.split(truncateMarker, 1).shift() as string;
```

Reply:
How about using the `!` operator?

```suggestion
  return fileString.split(truncateMarker, 1).shift()!;
```

Reply:
TIL.